### DISCUSSION TITLE
auth/login function

### DISCUSSION THREAD

**Discussion by zillionare** [h2oai/wave#1384] [Q&A] [answered]
I know we can start waved with keyclock so we can workaround this issue. However, if we have to implement auth/login by myself, how could I:

1. if user is not logged in, then redirect to /login page
2. at /login page, when user provide name/password and succeed in auth, redirect them to homepage.

Basically it's routing among different pages (not hashes, it's not intuitively routing among hashes, with regard to auth).

Thanks for help!

**Comment by lo5** [comment, marked as answer]
Apps in Wave don't have explicit control over this. Typically, you'd hook up the Wave server to OIDC, then throw one or more apps behind the Wave server. The Wave server interacts with the OIDC provider and presents the configured auth flow.
